design/cross-cutting-concepts.qmd

```diff
@@ -1,15 +1,148 @@
-# Cross-cutting concepts
+---
+title: Cross-cutting concepts
+---
 
 {{< include ../includes/_wip.qmd >}}
 
-## Privacy
+## Security
+
+There are a few main security concerns: software/code-based,
+authentication/permission-based, and server-based.
+
+-   **Software and code-based concerns**: They are the technical
+    decisions we make, the code we write, and the software we depend on.
+
+-   **Authentication and permission-based concerns**: They are related
+    to how permission levels are given, who can do what to the data, and
+    how users are authenticated.
+
+-   **Data-in-transit concerns**: These would be how secure the
+    connection is whenever data is transmitted across the Internet
+    between locations, such as with data entry.
+
+-   **Server-based concerns**: These are related mostly to data privacy,
+    discussed in detail in the [subsection](#privacy) below. Those
+    legally responsible for the security and privacy concerns at this
+    level are those who control the data and who manage the servers, and
+    so is outside the scope of Seedcase.
+
+-   Security layer diagram
+
+    @startuml
+    title Security layer diagram
+
+    rectangle "Authentication and Authorization" as auth
+    rectangle "API Security and Access" as api
+    rectangle "Application Logic and Data Validation" as logic
+    rectangle "Database Management System (DBMS)" as dbms
+    rectangle "Database Security" as db_security
+
+    auth --> api
+    api --> logic
+    logic --> dbms
+    dbms --> db_security
+    @enduml
```

Comment:
Sadly, plantuml can not be rendered within the document (unlike Mermaid, which can), so this needs to be in its own file.